spec/model_spec.cr

```diff
@@ -97,4 +120,48 @@ describe LuckyRecord::Model do
       QueryMe.column_names.should eq [:id, :created_at, :updated_at, :email, :age]
     end
   end
+
+  describe ".ensure_correct_field_mappings" do
+    # table is missing
+    it "raises on missing table" do
+      missing_table = MissingTable.new(1, Time.new, Time.new)
+      expect_raises Exception, "The table uusers was not found, did you mean users?" do
+        missing_table.ensure_correct_field_mappings!
+      end
+    end
+
+    # field defined in model without a matching column in table
+    it "raises on fields with missing columns" do
+      now = Time.now
+      user = MissingColumn.new id: 1,
+        created_at: now,
+        updated_at: now,
+        missing: "missing"
+      expect_raises Exception, "The table users does not have a 'missing' column. Make sure you've added it to the migration." do
+        user.ensure_correct_field_mappings!
+      end
+    end
+
+    it "raises on nilable fields with required columns" do
+      now = Time.now
+      user = OptionalFieldOnRequiredColumn.new id: 1,
```

Comment:
Now that I'm reading this in the test I think it might be more clear if you add `ModelWith` to these models. Otherwise it looks like you're instantiating a Column object and not a model